src/clib/pioc_support.c

```diff
@@ -93,19 +94,57 @@ int PIOc_strerror(int pioerr, char *errmsg)
 int PIOc_set_log_level(int level)
 {
 #if PIO_ENABLE_LOGGING
-    char log_filename[NC_MAX_NAME];
-
     /* Set the log level. */
     pio_log_level = level;
+#endif /* PIO_ENABLE_LOGGING */
+    return PIO_NOERR;
+}
 
-    /* Create a filename with the rank in it. */
-    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
-    sprintf(log_filename, "pio_log_%d.txt", my_rank);
+/** Initialize logging
+ * Open log file, if not opened yet, or
+ * increment ref count if already open
+ */
+void pio_init_logging(void )
```

Comment:
The old implementation looked hacky to me. PIOc_set_log_level() should not be initializing the logging, it should be setting the log level. The logging should be initialized when one initializes the pio library.
When are the log files closed in the prior impl? It also used invalid file handles for ops.
Also, Logging is always routed to stdout (even if the log files are not open yet).
If you want to improve on this impl, please feel free to create a new PR after this change is merged.